src/wobblySpinnerSheet.js

```diff
@@ -0,0 +1,72 @@
+import * as keyframesSheet from "./keyframesSheet";
+import wobblySpinnerRules from "./wobblySpinnerRules";
+import jss from "jss";
+import nested from "jss-nested";
+
+jss.use(nested()); // Allow for `&::after` in rules.
+
+const sheetRefCountMap = new Map(); // Map of JSS created `sheet` and `refCount`, indexed by `rules`.
+
+function attachSheet(sheet, rules) {
+    if (!sheet) {
+        sheet = jss.createStyleSheet(rules, {meta: "WobblySpinner"});
+    }
+    sheet.attach();
+    return sheet;
+}
+
+function detachSheet(sheet) {
+    sheet.detach();
+}
+
+function createRulesFromProps({diameter, thickness, wobbleOffset, spinRate, outerColor, innerColor, wobbleColor}) {
+    return wobblySpinnerRules({
+        thickness,
+        afterTopLeft: -thickness + wobbleOffset,
+        diameter,
+        spinRate,
+        wobbleRate: spinRate * 1.5,
+        borderRadius: (diameter + (thickness * 2)) / 2,
+        outerColor,
+        innerColor,
+        wobbleColor
+    });
+}
+
+export function arePropsEqual(props, nextProps) {
+    return props.diameter === nextProps.diameter &&
+        props.thickness === nextProps.thickness &&
+        props.spinRate === nextProps.spinRate &&
+        props.wobbleOffset === nextProps.wobbleOffset &&
+        props.outerColor === nextProps.outerColor &&
+        props.innerColor === nextProps.innerColor &&
+        props.wobbleColor === nextProps.wobbleColor;
+}
+
+export function mount(props) {
+    if (sheetRefCountMap.size === 0) {
+        keyframesSheet.mount();
+    }
+    var rules = createRulesFromProps(props);
+    var sheetRef = sheetRefCountMap.get(rules) || { refCount: 0 };
+    if (sheetRef.refCount === 0) {
+        sheetRef.sheet = attachSheet(sheetRef.sheet, rules); // If sheet doesn't exist, one will be created.
+    }
+    sheetRef.refCount++;
+    sheetRefCountMap.set(rules, sheetRef);
+    return { key: rules, className: sheetRef.sheet.classes.spinner };
+}
+
+export function unmount(rules) {
+    var sheetRef = sheetRefCountMap.get(rules);
+    sheetRef.refCount--;
+    if (sheetRef.refCount === 0) {
+        detachSheet(sheetRef.sheet);
+        sheetRefCountMap.delete(rules);
+        if (sheetRefCountMap.size === 0) {
+            keyframesSheet.unmount();
+        }
+    } else {
+        sheetRefCountMap.set(rules, sheetRef);
+    }
+}
```

Comment:
IMO, this module has two concerns. Managing the lifecycle of the stylesheet and the rules.